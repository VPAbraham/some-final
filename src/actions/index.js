export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const addOrders = orders => ({
  type: 'ADD_ORDERS',
  orders
})
