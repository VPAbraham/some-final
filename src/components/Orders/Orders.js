import React, { Component } from 'react';
import './Orders.css';
import { getOrders } from '../../apiCalls';
import { setOrders } from '../../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { deleteOrder } from '../../apiCalls';


class Orders extends Component {
  constructor(props) {
    super();
    this.props = props
  }

  async componentDidMount() {
    let { setOrders } = this.props
    try {
      const orderData = await getOrders()
      await console.log(orderData)
      await setOrders(orderData.orders)
    } catch (err) {
      console.error('Error fetching:', err)
    }
  }

  render() {

    const delOrderCard = (e) => {
      console.log(e)
      return e
    }

    const orderEls = this.props.orders.map(order => {
      return (
        <div key={Date.now()} className="order">
          <h3>{order.name}</h3>
          <ul className="ingredient-list">
            {order.ingredients.map(ingredient => {
              return <li>{ingredient}</li>
            })}
          </ul>
          <button onClick={(() => deleteOrder(order))}> Delete order </button>
        </div>
      )
    });


    return (
      <section>
        {orderEls.length ? orderEls : <p>No orders yet!</p>}
      </section>
    );
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setOrders,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Orders);