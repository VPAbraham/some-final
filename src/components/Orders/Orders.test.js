import React from 'react';
import {Orders, mapStateToProps, mapDispatchToProps} from './Orders';
import { shallow } from 'enzyme';
import { setOrders } from '../../actions'; 

describe('Orders', () => {
  let wrapper;

  const mockOrders = [{ id: 1, name: 'Robbie', ingredients: ['stuff1', 'stuff2'] },
    { id: 2, name: 'Travis', ingredients: ['stuff3', 'stuff4'] }];

  beforeEach(() => {
    wrapper = shallow(<Orders
      orders={mockOrders}
    /> )

  });
  
  it('should match the snapshot with the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
});