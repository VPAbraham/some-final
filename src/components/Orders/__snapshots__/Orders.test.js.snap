// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot with the data passed in 1`] = `
<section>
  <div
    className="order"
  >
    <h3>
      Robbie
    </h3>
    <ul
      className="ingredient-list"
    >
      <li>
        stuff1
      </li>
      <li>
        stuff2
      </li>
    </ul>
  </div>
  <div
    className="order"
  >
    <h3>
      Travis
    </h3>
    <ul
      className="ingredient-list"
    >
      <li>
        stuff3
      </li>
      <li>
        stuff4
      </li>
    </ul>
  </div>
</section>
`;
